partiql-ast/src/main/kotlin/org/partiql/ast/helpers/ToIon.kt

```diff
@@ -95,8 +96,21 @@ internal object ToIon : PartiQLValueBaseVisitor<IonElement, Unit>() {
     override fun visitBlob(v: BlobValue, ctx: Unit) = v.toIon { ionBlob(value) }
 
     override fun visitTimestamp(v: TimestampValue, ctx: Unit) = v.toIon {
-        val sql = java.sql.Timestamp.valueOf(value)
-        ionTimestamp(Timestamp.forSqlTimestampZ(sql))
+        val offset = when (val z = v.value.timeZone) {
+            TimeZone.UnknownTimeZone -> 0
+            is TimeZone.UtcOffset -> z.totalOffsetMinutes
+            null -> 0
+        }
+        val timestamp = Timestamp.forSecond(
```

Comment:
See this Javadoc note regarding unknown timezone: https://javadoc.io/doc/software.amazon.ion/ion-java/1.0.0/software/amazon/ion/Timestamp.html#forSecond-int-int-int-int-int-java.math.BigDecimal-java.lang.Integer-